### PR TITLE
refactor (app): Add evm basic transaction support

### PR DESCRIPTION
**Pull request by ujjwal-cyph**
Implements https://app.clickup.com/t/9002019994/PRF-4900

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4900 Ethereum send transaction support](https://app.clickup.com/t/9002019994/PRF-4900)

File: apps/evm_family/evm_sign_txn.c [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+                   ERROR_DATA_FLOW_INVALID_DATA);
+    status = false;
+  }
+
+  /* hard limit on the transaction size that can be fetched and kept in RAM.
+   * this limit can be removed once the RLP decoder can operate on running
+   * transaction buffer stream and get user confirmations on the go
+   */
+  // TODO: pending; currently ineffective
+  if (EVM_TRANSACTION_SIZE_CAP < txn_context->transaction_size) {
```
Comment (kartikCypherock) [inline apps/evm_family/evm_sign_txn.c:256]: How is `txn_context->transaction_size` populated?

File: apps/evm_family/evm_sign_txn.c [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+  }
+  const common_chunk_payload_chunk_t *chunk = &txn_data->chunk_payload.chunk;
+  txn_metadata dummy_metadata;
+  memcpy(txn_context->transaction, chunk->bytes, chunk->size);
+  // verify the received transaction
+  eth_byte_array_to_unsigned_txn(txn_context->transaction,
+                                 txn_context->transaction_size,
+                                 &txn_context->transaction_info,
+                                 &dummy_metadata);
+  eth_validate_unsigned_txn(&txn_context->transaction_info, &dummy_metadata);
```
Comment (kartikCypherock) [inline apps/evm_family/evm_sign_txn.c:334]: Return value left unchecked purposefully?

File: apps/evm_family/evm_sign_txn.c (hunk 1 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_sign_txn.c:256]: it's a todo. will be replaced by a member of `txn_context->init_info`

File: apps/evm_family/evm_sign_txn.c (hunk 2 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_sign_txn.c]: not-intentional. updated

File: apps/evm_family/evm_sign_txn.c [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+    }
+  }
+
+  // TODO: decide on handling blind signing
+  // show warning for unknown EVM function; take user consent
+  if (PAYLOAD_SIGNATURE_NOT_WHITELISTED ==
+      txn_context->transaction_info.payload_status) {
+    hd_path_array_to_string(hd_path, depth, false, address, sizeof(address));
+    if (!core_confirmation(UI_TEXT_BLIND_SIGNING_WARNING, evm_send_error) ||
+        !core_scroll_page("Verify Derivation Path", address, evm_send_error)) {
```
[the remark this answers is not in the document]
Comment (ujjwal-cyph) [reply apps/evm_family/evm_sign_txn.c]: moved

File: apps/evm_family/evm_sign_txn.c (hunk 1 above)
Comment (amanCypherock) [reply apps/evm_family/evm_sign_txn.c:256]: There are two ways to get the total size of the txn/msg data.

1. Current method: 
    From first chunk (total size = chunk size + remaining size)
2. Change in proto required:  
    Get the size from the initiate request from Sdk.

Comment (amanCypherock) [reply apps/evm_family/evm_sign_txn.c:256]: > it's a todo. will be replaced by a member of `txn_context->init_info`

Are you suggesting the 2nd method here?

Comment (ujjwal-cyph) [reply apps/evm_family/evm_sign_txn.c:256]: yes

Comment (ujjwal-cyph) [reply apps/evm_family/evm_sign_txn.c]: added all the necessary changes for this

File: apps/evm_family/eth/eth_app.h [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**
- * @file    eth_app.h
+ * @file    evm_app.h
```
Comment (kartikCypherock) [inline apps/evm_family/eth/eth_app.h:2]: Please undo this

File: apps/evm_family/evm_txn_helpers.h [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  uint8_t *payload;
+
+  uint8_t chain_id_size[1];
+  uint8_t chain_id[8];
+
+  uint8_t dummy_r[1];
+  uint8_t dummy_s[1];
+
+  PAYLOAD_STATUS payload_status;
+} eth_unsigned_txn;
```
Comment (amanCypherock) [inline apps/evm_family/evm_txn_helpers.h:84]: Would this differ fir other EVMs?
In that case can we rename as evm_unsigned_txn?

File: apps/evm_family/evm_txn_helpers.c [hunk 6]
```diff
@@ -0,0 +1,425 @@
...
+ */
+static uint64_t get_decode_length(const uint8_t *seq,
+                                  const uint64_t seq_len,
+                                  uint64_t *decoded_len,
+                                  seq_type *type);
+
+static PAYLOAD_STATUS eth_decode_txn_payload(
+    const eth_unsigned_txn *eth_utxn_ptr,
+    txn_metadata *metadata_ptr);
+
```
Comment (amanCypherock) [inline apps/evm_family/evm_txn_helpers.c:105]: Add doxygen comments

**Review by kartikCypherock — APPROVED** [review]

File: apps/evm_family/evm_txn_helpers.c (hunk 6 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_txn_helpers.c:105]: old functions relocated.

Comment (ujjwal-cyph) [reply apps/evm_family/evm_txn_helpers.c]: added

File: apps/evm_family/evm_txn_helpers.h (hunk 5 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_txn_helpers.h]: renamed

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**